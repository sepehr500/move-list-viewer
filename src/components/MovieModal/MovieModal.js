import React from "react";
import PropTypes from "prop-types";
import Modal from "react-modal";
import Rater from "react-rater";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";

const MovieModal = ({
  title,
  thumbnailPath,
  releaseDate,
  overview,
  score,
  onClose,
  onFavorite,
  isFavorite
}) => {
  return (
    <Modal
      isOpen={true}
      ariaHideApp={false}
      overlayClassName="Overlay"
      className="Modal"
      contentLabel="Example Modal"
    >
      <div className="flex-grow-1 center">
        <img alt={`${title} poster`} src={thumbnailPath} />
      </div>
      <div className="flex flex-column pa3 flex-grow-0 mw6">
        <div className="tc f3 mb3">
          <div>{title}</div>
        </div>
        <button
          data-testid="star-button"
          aria-label={isFavorite ? "Click to favorite" : "Click to unfavorite"}
          onClick={onFavorite}
          className="star-button dim"
        >
          <FontAwesomeIcon
            data-testid="star"
            icon={faStar}
            size="3x"
            className={`${isFavorite ? "gold" : "gray"} mb3 Slow-transition`}
          />
        </button>
        <div>
          <div className="mb3">Release date: {releaseDate}</div>
        </div>
        <div className="f1 pb1">
          <Rater interactive={false} total={5} rating={score} />
        </div>
        <div className="mb3">
          <div>{overview}</div>
        </div>
        <div>
          <button
            data-testid="closeButton"
            onClick={onClose}
            className="shadow-1 pb3 f3 w-100 br3 Details-button"
          >
            close
          </button>
        </div>
      </div>
    </Modal>
  );
};

MovieModal.propTypes = {
  title: PropTypes.string.isRequired,
  thumbnailPath: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  onClose: PropTypes.func.isRequired,
  onFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.bool.isRequired
};

export { MovieModal };
