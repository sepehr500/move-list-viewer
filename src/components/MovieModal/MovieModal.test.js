import React from "react";
import { mount } from "enzyme";
import { MovieModal } from "./MovieModal";

it("renders correctly", () => {
  const tree = mount(
    <MovieModal
      title="Gone with the wind"
      thumbnailPath="/asdf9n.jpg"
      releaseDate="2016-08-04"
      score={4}
      isFavorite
      overview="Some stuff happens"
      onFavorite={jest.fn()}
      onClose={jest.fn()}
      onClick={jest.fn()}
    />
  );
  expect(tree).toMatchSnapshot();
});
