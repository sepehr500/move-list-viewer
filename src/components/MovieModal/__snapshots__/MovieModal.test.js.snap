// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MovieModal
  isFavorite={true}
  onClick={[MockFunction]}
  onClose={[MockFunction]}
  onFavorite={[MockFunction]}
  overview="Some stuff happens"
  releaseDate="2016-08-04"
  score={4}
  thumbnailPath="/asdf9n.jpg"
  title="Gone with the wind"
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="Modal"
    closeTimeoutMS={0}
    contentLabel="Gone with the wind details"
    isOpen={true}
    overlayClassName="Overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        >
          <div
            class="ReactModal__Overlay ReactModal__Overlay--after-open Overlay"
          >
            <div
              aria-label="Gone with the wind details"
              class="ReactModal__Content ReactModal__Content--after-open Modal"
              role="dialog"
              tabindex="-1"
            >
              <img
                alt="Gone with the wind poster"
                class="center"
                src="/asdf9n.jpg"
                style="border-radius: 2%;"
              />
              <div
                class="pa3 mw6"
              >
                <div
                  class="tc f3 mb3"
                >
                  <h1
                    class="f2"
                  >
                    Gone with the wind
                  </h1>
                </div>
                <button
                  aria-label="Click to unfavorite"
                  class="star-button dim"
                  data-testid="star-button"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-star fa-w-18 fa-3x gold mb3 Slow-transition"
                    data-icon="star"
                    data-prefix="fas"
                    data-testid="star"
                    focusable="false"
                    role="img"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <div>
                  <p>
                    Release date: 
                    2016-08-04
                  </p>
                </div>
                <div
                  class="f1 pb1"
                >
                  <div
                    class="react-rater"
                  >
                    <div>
                      <div
                        class="react-rater-star is-disabled is-active"
                      >
                        ★
                      </div>
                    </div>
                    <div>
                      <div
                        class="react-rater-star is-disabled is-active"
                      >
                        ★
                      </div>
                    </div>
                    <div>
                      <div
                        class="react-rater-star is-disabled is-active"
                      >
                        ★
                      </div>
                    </div>
                    <div>
                      <div
                        class="react-rater-star is-disabled is-active"
                      >
                        ★
                      </div>
                    </div>
                    <div>
                      <div
                        class="react-rater-star is-disabled"
                      >
                        ★
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="mb3"
                >
                  <p>
                    Some stuff happens
                  </p>
                </div>
                <div>
                  <button
                    class="shadow-1 pb3 f3 w-100 br3 Details-button"
                    data-testid="closeButton"
                  >
                    close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalPortal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        className="Modal"
        closeTimeoutMS={0}
        contentLabel="Gone with the wind details"
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={true}
        overlayClassName="Overlay"
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {},
            "overlay": Object {},
          }
        }
      >
        <div
          className="ReactModal__Overlay ReactModal__Overlay--after-open Overlay"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={Object {}}
        >
          <div
            aria-label="Gone with the wind details"
            className="ReactModal__Content ReactModal__Content--after-open Modal"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="dialog"
            style={Object {}}
            tabIndex="-1"
          >
            <img
              alt="Gone with the wind poster"
              className="center"
              src="/asdf9n.jpg"
              style={
                Object {
                  "borderRadius": "2%",
                }
              }
            />
            <div
              className="pa3 mw6"
            >
              <div
                className="tc f3 mb3"
              >
                <h1
                  className="f2"
                >
                  Gone with the wind
                </h1>
              </div>
              <button
                aria-label="Click to unfavorite"
                className="star-button dim"
                data-testid="star-button"
                onClick={[MockFunction]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="gold mb3 Slow-transition"
                  data-testid="star"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        576,
                        512,
                        Array [],
                        "f005",
                        "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
                      ],
                      "iconName": "star",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="3x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-star fa-w-18 fa-3x gold mb3 Slow-transition"
                    data-icon="star"
                    data-prefix="fas"
                    data-testid="star"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </button>
              <div>
                <p>
                  Release date: 
                  2016-08-04
                </p>
              </div>
              <div
                className="f1 pb1"
              >
                <Rater
                  interactive={false}
                  rating={4}
                  total={5}
                >
                  <div
                    className="react-rater"
                  >
                    <div
                      key="star-0"
                      onClick={null}
                      onMouseOver={null}
                    >
                      <Star
                        isActive={true}
                        isActiveHalf={false}
                        isDisabled={true}
                        willBeActive={false}
                      >
                        <div
                          className="react-rater-star is-disabled is-active"
                        >
                          ★
                        </div>
                      </Star>
                    </div>
                    <div
                      key="star-1"
                      onClick={null}
                      onMouseOver={null}
                    >
                      <Star
                        isActive={true}
                        isActiveHalf={false}
                        isDisabled={true}
                        willBeActive={false}
                      >
                        <div
                          className="react-rater-star is-disabled is-active"
                        >
                          ★
                        </div>
                      </Star>
                    </div>
                    <div
                      key="star-2"
                      onClick={null}
                      onMouseOver={null}
                    >
                      <Star
                        isActive={true}
                        isActiveHalf={false}
                        isDisabled={true}
                        willBeActive={false}
                      >
                        <div
                          className="react-rater-star is-disabled is-active"
                        >
                          ★
                        </div>
                      </Star>
                    </div>
                    <div
                      key="star-3"
                      onClick={null}
                      onMouseOver={null}
                    >
                      <Star
                        isActive={true}
                        isActiveHalf={false}
                        isDisabled={true}
                        willBeActive={false}
                      >
                        <div
                          className="react-rater-star is-disabled is-active"
                        >
                          ★
                        </div>
                      </Star>
                    </div>
                    <div
                      key="star-4"
                      onClick={null}
                      onMouseOver={null}
                    >
                      <Star
                        isActive={false}
                        isActiveHalf={false}
                        isDisabled={true}
                        willBeActive={false}
                      >
                        <div
                          className="react-rater-star is-disabled"
                        >
                          ★
                        </div>
                      </Star>
                    </div>
                  </div>
                </Rater>
              </div>
              <div
                className="mb3"
              >
                <p>
                  Some stuff happens
                </p>
              </div>
              <div>
                <button
                  className="shadow-1 pb3 f3 w-100 br3 Details-button"
                  data-testid="closeButton"
                  onClick={[MockFunction]}
                >
                  close
                </button>
              </div>
            </div>
          </div>
        </div>
      </ModalPortal>
    </Portal>
  </Modal>
</MovieModal>
`;
